integrations/gitlab/gitlab_integration/ocean.py

```diff
@@ -160,6 +163,41 @@ async def resync_folders(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
                             yield folders_batch
 
 
+@ocean.on_resync(ObjectKind.FILE)
+async def resync_files(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    for service in get_cached_all_services():
+        gitlab_resource_config: GitLabFilesResourceConfig = typing.cast(
+            "GitLabFilesResourceConfig", event.resource_config
+        )
+        if not isinstance(gitlab_resource_config, GitLabFilesResourceConfig):
+            return
+
+        selector = gitlab_resource_config.selector
+        async for projects_batch in service.get_all_projects():
+
+            ## If user does not provide any path, we won't be able to fetch files since this can be a very expensive operation going through all the files in all the projects
+            if selector.files and selector.files.path:
+                if (
+                    selector.files.repos
+                ):  ## If user provides repos, we will only fetch files from those repos
+                    for project in projects_batch:
+                        if project.name in selector.files.repos:
+                            async for files_batch in service.get_all_files_in_project(
+                                project, selector.files.path
+                            ):
+                                yield files_batch
+                else:  ## If user does not provide repos, we will fetch files from all the projects
+                    for project in projects_batch:
+                        async for files_batch in service.get_all_files_in_project(
+                            project, selector.files.path
+                        ):
+                            yield files_batch
```

Comment:
changing the structure a bit can make the code more readable. 
here is my suggestion:

```suggestion
               for project in projects_batch:
                   if project_matches_path_filter(project):
                        async for files_batch in service.get_all_files_in_project(
                                project, selector.files.path
                            ):
                                yield files_batch
```

the `project_matches_path_filter` will contain the logic of whether to filter the project or not